### PR TITLE
Japanese language support

### PR DESCRIPTION
**Pull request by Taka005**
Addition of Japanese localization files

### CODE REVIEW

File: BDArmory/Distribution/GameData/BDArmory/Localization/UI/ja.cfg [hunk 1]
```diff
@@ -0,0 +1,1194 @@
...
+        #LOC_BDArmory_AIWindow_ExtendDistanceAirToGroundGuns_Context = 空対地距離の延長 (砲)
+        #LOC_BDArmory_AIWindow_ExtendDistanceAirToGround_Context = 空対地距離の延長
+        #LOC_BDArmory_AIWindow_Extendvel = ターゲットが遅すぎて方向を向くことができない場合に設定します
+        #LOC_BDArmory_AIWindow_ExtendAngle = ターゲットが回転半径の外側にあるときに設定します
+        #LOC_BDArmory_AIWindow_ExtendDist = ターゲットが近すぎて振り向くことができない場合に設定します
+        #LOC_BDArmory_AIWindow_ExtendAbortTimeContext = 延長中に距離を稼げなかった場合は時間を中止します。
+        #LOC_BDArmory_AIWindow_terrainMin = 理想的な機体の向きを実現する回転半径乗数
+        #LOC_BDArmory_AIWindow_terrainMax = 反転したクラフトの向きの回転半径乗数
+        #LOC_BDArmory_AIWindow_InvertedTerrainAvoidanceCriticalAngleContext = 逆地形回避またはローリングファーストの臨界角
+        #LOC_BDArmory_AIWindow_TerrainAvoidanceControlSurfaceDeploymentTimeContext = コントロール サーフェスの展開にかかる時間の見積もり
```
Comment (BrettRyland) [inline BDArmory/Distribution/GameData/BDArmory/Localization/UI/ja.cfg:763]: `#LOC_BDArmory_AIWindow_TerrainAvoidanceControlSurfaceDeploymentTimeContext` is now `#LOC_BDArmory_AIWindow_TerrainAvoidanceVesselReactionTimeContext = Estimate of time required to setup for optimal turning`

File: BDArmory/Distribution/GameData/BDArmory/Localization/UI/ja.cfg [hunk 2]
```diff
@@ -0,0 +1,1194 @@
...
+        #LOC_BDArmory_AIWindow_Avoidance = 血管の回避
+        #LOC_BDArmory_AIWindow_CollisionAvoidanceThreshold = 距離のしきい値
+        #LOC_BDArmory_AIWindow_CollisionAvoidanceLookAheadPeriod = 先読み時間
+        #LOC_BDArmory_AIWindow_CollisionAvoidanceStrength = 応答強度
+        #LOC_BDArmory_AIWindow_StandoffDistance = スタンドオフ距離
+        #LOC_BDArmory_AIWindow_Terrain = 地形
+        #LOC_BDArmory_AIWindow_TurnRadiusMin = 地形回避最小値
+        #LOC_BDArmory_AIWindow_TurnRadiusMax = 地形回避最大値
+        #LOC_BDArmory_AIWindow_InvertedTerrainAvoidanceCriticalAngle = 逆クリティカル。角度
+        #LOC_BDArmory_AIWindow_TerrainAvoidanceControlSurfaceDeploymentTime = Ctrl Srf 導入時間
```
Comment (BrettRyland) [inline BDArmory/Distribution/GameData/BDArmory/Localization/UI/ja.cfg:815]: `#LOC_BDArmory_AIWindow_TerrainAvoidanceControlSurfaceDeploymentTime` is now `#LOC_BDArmory_AIWindow_TerrainAvoidanceVesselReactionTime = Vessel Reaction Time`

File: BDArmory/Distribution/GameData/BDArmory/Localization/UI/ja.cfg [hunk 3]
```diff
@@ -0,0 +1,1194 @@
...
+        #LOC_BDArmory_BDAScores_Title = トーナメントスコア
+        #LOC_BDArmory_BDAScores_Weights = スコアの量
+        #LOC_BDArmory_BDAScores_Round = ラウンド
+        #LOC_BDArmory_BDAScores_Heat = 体力
+
+        // Custom Spawn Templates
+        #LOC_BDArmory_Settings_CustomSpawnTemplateOptions = スポーンテンプレート設定
+        #LOC_BDArmory_Settings_SpawnOnly = スポーンのみ
+        #LOC_BDArmory_Settings_SpawnAndStartCompetition = スポーンして対戦を開始
+        #LOC_BDArmory_Settings_CustomSpawnTemplate_Refresh = リロード
```
Comment (BrettRyland) [inline BDArmory/Distribution/GameData/BDArmory/Localization/UI/ja.cfg:399]: `#LOC_BDArmory_Settings_CustomSpawnTemplate_Refresh` isn't used anymore, `#LOC_BDArmory_CraftBrowser_Refresh` is used instead.

`#LOC_BDArmory_Settings_CustomSpawnTemplate_ReplaceTeam = Replace Teams` is missing, which is used to replace all the slots on a team in a custom spawn template with the selected craft.

**Review by BrettRyland — COMMENTED** [review]
Excellent! Thanks for adding these.
There is a typo and a couple of the localisation tags have changed, but otherwise it looks quite thorough.

**Comment by Taka005** [comment]
All fixed.

I have a question, what is the cause of the fact that I created a Japanese language file based on an English language file, but it is still partially in English on KSP?

P.S.
I don't know much English so I'm a little wrong

File: BDArmory/Distribution/GameData/BDArmory/Localization/UI/ja.cfg [hunk 4]
```diff
@@ -760,8 +760,8 @@ Localization
         #LOC_BDArmory_AIWindow_ExtendAbortTimeContext = 延長中に距離を稼げなかった場合は時間を中止します。
         #LOC_BDArmory_AIWindow_terrainMin = 理想的な機体の向きを実現する回転半径乗数
         #LOC_BDArmory_AIWindow_terrainMax = 反転したクラフトの向きの回転半径乗数
-        #LOC_BDArmory_AIWindow_InvertedTerrainAvoidanceCriticalAngleContext = 逆地形回避またはローリングファーストの臨界角
-        #LOC_BDArmory_AIWindow_TerrainAvoidanceControlSurfaceDeploymentTimeContext = コントロール サーフェスの展開にかかる時間の見積もり
+        ##LOC_BDArmory_AIWindow_TerrainAvoidanceVesselReactionTimeContext = 逆地形回避またはローリングファーストの臨界角
```
Comment (BrettRyland) [inline BDArmory/Distribution/GameData/BDArmory/Localization/UI/ja.cfg:763]: `#LOC_BDArmory_AIWindow_InvertedTerrainAvoidanceCriticalAngle` should not have been removed here and you have an extra `#` in `#LOC_BDArmory_AIWindow_TerrainAvoidanceVesselReactionTimeContext`.
Looking at google translate,
Screenshot_20230826_170941
I think you've mixed up the tags. This line ought to be
```
#LOC_BDArmory_AIWindow_InvertedTerrainAvoidanceCriticalAngleContext = 逆地形回避またはローリングファーストの臨界角
```
and the line below ought to be something like
```
#LOC_BDArmory_AIWindow_TerrainAvoidanceVesselReactionTimeContext = 最適な旋回のためのセットアップに必要な時間の見積もり
```
(note: I've just used google translate for this, so I'm not sure of the translation). The original explanation for this value was "the time required to deploy the control surfaces", but that got changed to "an estimate of the time required to set up for optimal turning" after we realised that it was more involved than just adjusting control surfaces. It corresponds to the initial period in the diagram for "terrain threat distance" https://github.com/BrettRyland/BDArmory/wiki/2.1-AI-Tuning#terrain-avoidance labelled with "Adjusting control surfaces" where the plane continues in a roughly straight line (slightly S-shaped) before being able to properly turn in a circle.

The short version for the "vessel reaction time" still needs adjusting below on line 816.

**Comment by BrettRyland** [comment]
> I have a question, what is the cause of the fact that I created a Japanese language file based on an English language file, but it is still partially in English on KSP?

According to https://github.com/Apofoo/KSP-Localization/blob/master/README.md and https://steamcommunity.com/app/220200/discussions/0/1652169858530316480/, it depends on where you bought KSP. Testing this with the steam version, I find that changing the language in the properties tab of KSP in steam, it changes the language in buildID.txt (or buildID64.txt), but you also need to change it in KSP's settings.cfg file for all the localisation to work properly.
There may also be some parts of some mods that simply don't have localisation.

**Comment by Taka005** [comment]
I think this fixes everything.

**Comment by BrettRyland** [comment]
> I think this fixes everything.

Not quite, but I have enough to be able to make the remaining fixes myself before merging this.
